<launch>
  <!-- Launch file for velodyne16 VLP-16 LiDAR -->
    <rosparam command="load" file="$(find gslivm)/config/ntu.yaml" />
    <rosparam command="load" file="$(find gslivm)/config/basic_common.yaml" />

    <arg name="gnss_topic" default="/gps/fix" />
    <arg name="gnss_pose_topic" default="/gnss_pose" />
    <arg name="gnss_covariance" default="5.0" />

    <node pkg="gslivm" type="gnss_adapter" name="gnss_adapter" output="screen" />

    <param name="common/gnss_topic" value="$(arg gnss_topic)" />
    <param name="common/gnss_pose_topic" value="$(arg gnss_pose_topic)" />
    <param name="common/gnss_covariance" value="$(arg gnss_covariance)" />

    <param name="debug_output" type="bool" value="0"/>
    <param name="output_path" type="string" value="$(find gslivm)/output"/>
    <node pkg="gslivm" type="livo_node" name="livo_node" output="screen" /> 

    <!-- <arg name="bag_path" value="-s 10 /home/xieys/Downloads/bags/ntu/eee_03.bag" /> -->
    <arg name="bag_path" value="-s 10 /home/xieys/Downloads/bags/ntu/eee_02/eee_02.bag" />
    <!-- <arg name="bag_path" value="-s 1 /home/xieys/Downloads/bags/ntu/nya_01/nya_01.bag" /> -->
    <!-- <arg name="bag_path" value="-s 20 /home/xieys/Downloads/bags/ntu/sbs_01/sbs_01.bag" /> -->

    <node pkg="rosbag" type="play" name="player" output="screen" args="-q -d 3 $(arg bag_path)"/>
    
    <param name="bag_path" value="$(arg bag_path)"/>
    <node pkg="gslivm" name="odom_listener" type="listen_odom.py"  output="screen" />
</launch>


